lib/logstash/inputs/cloudwatch_logs.rb

```diff
@@ -201,78 +189,79 @@ def process_group(group)
     end
     @priority.delete(group)
     @priority << group
-  end #def process_group
+  end
+
+  def should_fetch_tags(log_group_name)
+    # only fetch tags if
+    # - there is no timestamp for when the tags were last updated
+    # - OR the tags were last updated more than an hour ago (60 seconds * 60 minutes)
+    @tag_cache[log_group_name][:last_updated].nil? ||
+      ((Time.now - @tag_cache[log_group_name][:last_updated]) > (60 * 60))
+  end
 
   def fetch_tags(log_group_name)
-    if @tag_cache.key?(log_group_name)
-      return @tag_cache[log_group_name][:tags]
-    else
-      tags = fetch_tags_from_cloudwatch(log_group_name)
-      @tag_cache[log_group_name] = { tags: tags}
-      return tags
-    end
+    return @tag_cache[log_group_name][:tags] if @tag_cache.key?(log_group_name) && !should_fetch_tags(log_group_name)
+
+    tags = fetch_tags_from_cloudwatch(log_group_name)
+    @tag_cache[log_group_name] = { tags: tags, last_updated: Time.now }
+    tags
```

Comment:
These are the only meaningful changes in the PR. The rest are formatting changes suggested by Rubocop which I do think makes the code more readable and maintainable